src/main/java/com/zendesk/maxwell/schema/MysqlSavedSchema.java

```diff
@@ -226,19 +226,30 @@ private void executeColumnInsert(Connection conn, ArrayList<Object> columnData)
 		columnData.clear();
 	}
 
-	public static MysqlSavedSchema restore(Connection connection, MaxwellContext context) throws SQLException, IOException, InvalidSchemaError {
-		MysqlSavedSchema s = new MysqlSavedSchema(context.getServerID(), context.getCaseSensitivity());
+	public static MysqlSavedSchema restore(MaxwellContext context, BinlogPosition targetPosition) throws SQLException, InvalidSchemaError {
+		try ( Connection conn = context.getMaxwellConnection() ) {
+			Long schemaID = findSchema(conn, targetPosition, context.getServerID());
+			if (schemaID == null)
+				return null;
```

Comment:
(so where we'd raise before, I think it'd be nice now do logger.error or warn, explaining in what kind of invalid state we were and that we recovered)